Source/CBL_SQLiteStorage.m

```diff
@@ -677,11 +677,18 @@ - (SInt64) _getDocNumericID: (NSString*)docID {
 - (SInt64) _createDocNumericID: (NSString*)docID {
     if (![_fmdb executeUpdate: @"INSERT OR IGNORE INTO docs (docid) VALUES (?)", docID])
         return -1;
-    if (_fmdb.changes == 0)
-        return -1;
+    if (_fmdb.changes == 0) {
+      SInt64 result = [self _getDocNumericID:docID];
```

Comment:
Actually the only change that's needed here is to replace `return -1` with `return 0`. The caller (-createOrGetDocNumericID:) will do the right thing if 0 is returned.